orangecontrib/text/widgets/ownyt.py

```diff
@@ -132,42 +132,48 @@ def __init__(self):
                                         focusPolicy=Qt.NoFocus)
 
     def new_query_input(self):
-        if self.running:
-            self.stop()
+        self.search.stop()
         self.search()
 
     def start_stop(self):
-        if self.running:
-            self.stop()
+        if self.search.running:
+            self.search.stop()
         else:
             self.query_box.synchronize(silent=True)
             self.search()
 
     @gui_require('nyt_api', 'no_api')
     @gui_require('recent_queries', 'no_query')
-    @asynchronous(allow_partial_results=True)
-    def search(self, on_progress, should_break):
+    def run_search(self):
```

Comment:
Same comment as below. Can we get rid of this `run_search` method and move the all `@gui_require` decorators directly on `search`?